public/js/resource.js

```diff
@@ -41,6 +40,123 @@ const linkUpload = () => {
   uploadItem(body);
 };
 
-// html tags for each form (add, update, or delete)
+const deleteResource = async (event) => {
+  event.preventDefault();
+  const { space_id, idea_id } = getContext();
+  const resource_id = $('.delete-resource').attr('data-id');
+
+  // markdown, image, or link data sent through.
+  const response = await fetch(
+    `/api/space/${space_id}/idea/${idea_id}/resource/${resource_id}`,
+    {
+      method: 'DELETE',
+      headers: { 'Content-Type': 'application/json' },
+    }
+  );
+
+  if (response.ok) {
+    localStorage.setItem('toast', `Resource removed`);
+    document.location.href = `/space/${space_id}/idea/${idea_id}`;
+  } else {
+    localStorage.setItem('toast', `Unable to delete this :().`);
+    toastIt(true);
+  }
+};
+
+const editLink = async (event) => {
+  event.preventDefault();
+  const { space_id, idea_id } = getContext();
+  const resource_id = $('.link-update').attr('data-id');
+  const body = {
+    id: resource_id,
+    name: $('.link-name').val(),
+    type: 'link',
+    content: $('.link-content').val(),
+  };
+  // markdown, image, or link data sent through.
+  const response = await fetch(
+    `/api/space/${space_id}/idea/${idea_id}/resource/${resource_id}`,
+    {
+      method: 'PUT',
+      body: JSON.stringify(body),
+      headers: { 'Content-Type': 'application/json' },
+    }
+  );
+
+  if (response.ok) {
+    localStorage.setItem('toast', `Resource updated`);
+    document.location.href = `/space/${space_id}/idea/${idea_id}`;
+  } else {
+    localStorage.setItem('toast', `Unable to update this :().`);
+    toastIt(true);
+  }
+};
+
+const editMarkdown = async (event) => {
+  event.preventDefault();
+  const { space_id, idea_id } = getContext();
+  const resource_id = $('.markdown-update').attr('data-id');
+
+  const body = {
+    id: resource_id,
+    name: $('.markdown-name').val(),
+    type: 'markdown',
+    content: $('.markdown-content').val(),
+  };
+  // markdown, image, or link data sent through.
+  const response = await fetch(
+    `/api/space/${space_id}/idea/${idea_id}/resource/${resource_id}`,
+    {
+      method: 'PUT',
+      body: JSON.stringify(body),
+      headers: { 'Content-Type': 'application/json' },
+    }
+  );
+
+  if (response.ok) {
+    localStorage.setItem('toast', `Resource updated`);
+    document.location.href = `/space/${space_id}/idea/${idea_id}`;
+  } else {
+    localStorage.setItem('toast', `Unable to update this :().`);
+    toastIt(true);
+  }
+};
+
+const currentResource = async (event) => {
+  event.preventDefault();
+  event.currentTarget;
+
+  const { space_id, idea_id } = getContext();
+  const resource_id = $('.editing-buttons').attr('data-id');
```

Comment:
$('.editing-buttons') will return an array of elements that match the editing-buttons class, so by default, jQuery will grab the first one in the list to chain off of--that's why you're getting that first one on the .attr() call afterwards.

Try passing the exact resource id you want with the event target from above.